src/frontends/pytorch/src/transforms/irfftn_complex_replacer.cpp

```diff
@@ -0,0 +1,142 @@
+            dim = std::make_shared<v0::Convert>(concat_list_construct(irfftn_op->input_value(2)), element::i32);
+            dim = std::make_shared<v8::Gather>(dim_range, dim, const_0);
+        } else if (!s_use_default) {
+            auto s_len = std::make_shared<v3::ShapeOf>(raw_s_input_maybe, element::i32);
+            auto range_start = std::make_shared<v1::Subtract>(input_rank, s_len);
+            auto range_start_scalar = std::make_shared<v0::Squeeze>(range_start);
+            dim = std::make_shared<v0::Range>(range_start_scalar, input_rank_scalar, const_scalar_1);
+        } else {
+            dim = std::make_shared<v0::Range>(const_scalar_0, input_rank_scalar, const_scalar_1);
+        }
+
+        auto default_s_raw = std::make_shared<v8::Gather>(complex_input_shape, dim, const_0);
+        auto last_s = std::make_shared<v8::Gather>(default_s_raw, const_neg_1, const_0);
+        auto last_s_m_1 = std::make_shared<v1::Subtract>(last_s, const_1);
+        auto s_upd = std::make_shared<v1::Multiply>(last_s_m_1, const_2);
+        auto s_shape = std::make_shared<v3::ShapeOf>(default_s_raw, element::i32);
+        auto last_s_idx = std::make_shared<v1::Subtract>(s_shape, const_1);
+        auto default_s = std::make_shared<v3::ScatterUpdate>(default_s_raw, last_s_idx, s_upd, const_0);
+
+        std::shared_ptr<ov::Node> s;
+        if (!s_use_default) {
+            auto full_s_cond = std::make_shared<v1::Equal>(raw_s_input_maybe, const_neg_1);
+            s = std::make_shared<v1::Select>(full_s_cond, default_s, raw_s_input_maybe);
+        } else {
+            s = default_s;
+        }
+
+        std::string norm;
+        if (const auto& fw_node_mode = std::dynamic_pointer_cast<ov::op::util::FrameworkNode>(
+                irfftn_op->input_value(3).get_node_shared_ptr())) {
+            const auto& attrs = fw_node_mode->get_attrs();
+            if (attrs.find("string_value") != attrs.end()) {
+                norm = attrs.at("string_value");
+            } else {
+                norm = "backward";
+            }
+        } else {
+            return false;
+        }
+
+        auto irdft = std::make_shared<v9::IRDFT>(input, dim, s);
+
+        auto n_int = std::make_shared<v1::ReduceProd>(s, const_0);
+        auto n = std::make_shared<v1::ConvertLike>(n_int, irdft);
+        std::shared_ptr<ov::Node> normalized_irfftn;
+        if (norm == "forward") {
+            normalized_irfftn = std::make_shared<v1::Multiply>(irdft, n);
+        } else if (norm == "backward") {
+            normalized_irfftn = irdft;
+        } else if (norm == "ortho") {
+            auto sqrt_n = std::make_shared<v0::Sqrt>(n);
+            normalized_irfftn = std::make_shared<v1::Multiply>(irdft, sqrt_n);
+        } else {
+            return false;
+        }
+
+        copy_runtime_info({irfftn_op, fw_node_complex_input}, normalized_irfftn);
+        replace_node(irfftn_op, normalized_irfftn);
```

Comment:
Please also preserve friendly name of the initial operation
`normalized_irfftn->set_friendly_name(irfftn_op->get_friendly_name());`

Reply:
Done